src/edfi-paging-test/edfi_paging_test/helpers/api_metadata.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from functools import cache
+from typing import Any, List, Dict
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+@cache
+def get_base_api_response(api_base_url: str) -> Dict[str, Any]:
+    """
+    Gets the metadata response from the base API. Cached to avoid multiple requests.
+
+    Parameters
+    ----------
+    api_base_url : str
+        The base URL of the API.
+
+    Returns
+    -------
+    Dict
+        The API metadata response from the server
+    """
+    logger.debug("Getting metadata from the api root.")
+    return requests.get(api_base_url).json()
+
+
+@cache
+def get_openapi_metadata_response(api_base_url: str) -> List[Dict[str, str]]:
+    """
+    Gets the OpenAPI metadata response from the base API. Cached to avoid multiple
+    requests.
+
+    Parameters
+    ----------
+    api_base_url : str
+        The base URL of the API.
+
+    Returns
+    -------
+    Dict
+        The OpenAPI metadata response from the server
+    """
+    logger.debug("Getting OpenAPI metadata from the api.")
+    base_api_response: Dict[str, Dict[str, str]] = get_base_api_response(api_base_url)
+    return requests.get(base_api_response["urls"]["openApiMetadata"]).json()
```

Comment:
This questions should have occurred to me with an earlier PR... what kind of error message are we going to get if we have the wrong base URL? Can we make sure that there is something meaningful instead of a `KeyError` or other scary message?

Reply:
Good idea. Added an "invalid url" message.